jena-arq/src/main/java/org/apache/jena/riot/system/Checker.java

```diff
@@ -206,36 +217,45 @@ public static boolean checkLiteral(String lexicalForm, String lang, ErrorHandler
 
     public static boolean checkLiteral(String lexicalForm, String lang, RDFDatatype datatype, ErrorHandler errorHandler, long line,
                                        long col) {
-        boolean hasLang = lang != null && !lang.equals("");
-        if ( !hasLang ) {
-            // Datatype check (and RDF 1.0 simple literals are always well formed)
-            if ( datatype != null )
-                return validateByDatatype(lexicalForm, datatype, errorHandler, line, col);
-            return true;
-        }
-
-        // Has a language.
-        if ( JenaRuntime.isRDF11 ) {
-            if ( datatype != null && !Objects.equals(datatype.getURI(), NodeConst.rdfLangString.getURI()) ) {
-                errorHandler(errorHandler).error("Literal has language but wrong datatype", line, col);
-                return false;
+        boolean hasLang = ( lang != null && !lang.isEmpty() );
+        boolean hasDatatype = datatype != null;
+
+        // Has language...
+        if ( hasLang ) {
+            // Has datatype (generally, a problem)...
```

Comment:
The comment implies it is common but as node are normally created by the parsers, it isn't. 

"If there is a language, the data type must be rdf:langString."

Reply:
Yeah, bad wording on my part.  I'll update the commenting throughout checkLiteral().